platform/generic_unix/wayland/os_wayland.cpp

```diff
@@ -487,8 +487,14 @@ int OS_Wayland::get_screen_dpi(int p_screen) const {
 	return outputs[p_screen]->scale * 96;
 }
 
+OS_GenericUnix *OS_Wayland::detect() {
+	if (!getenv("WAYLAND_DISPLAY") && !getenv("WAYLAND_SOCKET")) {
```

Comment:
Strictly speaking, a better solution would be to actually attempt to establish a connection with the display (ditto for X11), but that would move this task much earlier in the startup process than it was before. Open to feedback on this approach.